app/models/update.server.ts

```diff
@@ -114,3 +115,78 @@ export async function addSeeAlso(data: { [k: string]: FormDataEntryValue }) {
     },
   })
 }
+
+const EditingStatusMap: Record<EditingStatusType, string> = {
+  [EditingStatusType.FIRST_DRAFT]: "first_draft",
+  [EditingStatusType.REVISED_DRAFT]: "revised_draft",
+  [EditingStatusType.SEMANT_REVISED]: "semantically_revised",
+  [EditingStatusType.EDITED_FOR_STYLE]: "edited_for_style",
+  [EditingStatusType.CHIEF_EDITOR_OK]: "chief_editor_ok",
+  [EditingStatusType.NO_CDN_SUSP]: "no_cdn_susp",
+  [EditingStatusType.NO_CDN_CONF]: "no_cdn_conf",
+  [EditingStatusType.COPY_EDITED]: "final_proofing", // TODO: Confirm this with old codebase
+  [EditingStatusType.PROOF_READING]: "proofread", // TODO: Also confirm this with above
+}
+
+export async function updateEditingStatus(data: {
+  [k: string]: FormDataEntryValue
+}) {
+  const headword = getStringFromFormInput(data.headword)
+
+  // clear all values, because checkbox values aren't passed when not on
+  await resetAllEditingStatusValues(headword)
+  for (const key in data) {
+    await updateSingleEditingStatus(headword, key, data[key])
+  }
+}
+
+async function resetAllEditingStatusValues(headword: string) {
+  for (const key in EditingStatusMap) {
+    console.log(key)
+    await updateSingleEditingStatus(headword, key, "off")
+  }
+}
+
+async function updateSingleEditingStatus(
+  headword: string,
+  editingStatus: string,
+  value: FormDataEntryValue
+) {
+  const fieldName = EditingStatusMap[editingStatus as EditingStatusType]
+  const bool = getCheckboxValueAsBoolean(value)
+
+  // no-op if fieldname somehow doesn't exist
+  if (!fieldName) return null
+
+  await prisma.entry.update({
+    where: {
+      headword: headword,
+    },
+    data: {
+      [fieldName]: bool,
+    },
+  })
+}
+
+export async function updateEditingTools(data: {
+  [k: string]: FormDataEntryValue
+}) {
+  const headword = getStringFromFormInput(data.headword)
+  const isPublic = getCheckboxValueAsBoolean(data.isPublic)
+  const isLegacy = getCheckboxValueAsBoolean(data.isLegacy)
+
+  await prisma.entry.update({
+    where: {
+      headword: headword,
+    },
+    data: {
+      is_public: isPublic,
+      is_legacy: isLegacy,
+    },
+  })
+}
+
+function getCheckboxValueAsBoolean(value: FormDataEntryValue) {
```

Comment:
Let's move something like this into a utility file. I imagine it'll be useful across the codebase.